test/Main.purs

```diff
@@ -37,7 +37,7 @@ testRoundTrip
 testRoundTrip x = do
   let json = unsafeStringify (write x)
   log json
-  case runExcept (readJSON json) of
+  case runExcept (parseJSON json >>= read) of
```

Comment:
my only complaint is that i think parseJSON json >>= read will be so common that i'd like to have readJSON be put back in...